### DISCUSSION TITLE
Question around ambiguous relations

### DISCUSSION THREAD

**Discussion by dgrcode** [prisma/prisma#14345] [Q&A] [answered]
After reading the docs, which are amazing 🙌 , I still have a doubt regarding ambiguous relations. I'm quite new both with Prisma and relational databases, so maybe there's a simpler way of doing what I want but approaching the schema differently.

The general context is that I have an app where people can create charts, where each one has two fields: one representing the x coordinate, and another one representing the y.

I have the following schema:
```
model Field {
  uuid         String    @id @default(uuid())
  label        String
  type         FieldType
}

model Chart {
  uuid     String        @id @default(uuid())
  type     ChartType
  xField     Field @relation(fields: [xFieldUuid], references: [uuid])
  xFieldUuid String
  yField     Field @relation(fields: [yFieldUuid], references: [uuid])
  yFieldUuid String
}
```

I'm getting the following error:
```
Error validating model "Chart": Ambiguous relation detected. The fields `xField` and `yField` in model `Chart` both refer to `Field`. Please provide different relation names for them by adding `@relation(<name>).
```

I understand from the docs that I need to add two new props to the Field model, so each of the relation in the Chart point to one of them. So the following should work (I've actually checked that it does):
```
model Field {
  uuid         String    @id @default(uuid())
  label        String
  type         FieldType
  chartsAsX Chart[] @relation("xField")
  chartsAsY Chart[] @relation("yField")
}

model Chart {
  uuid     String        @id @default(uuid())
  type     ChartType
  xField     Field @relation("xField", fields: [xFieldUuid], references: [uuid])
  xFieldUuid String
  yField     Field @relation("yField", fields: [yFieldUuid], references: [uuid])
  yFieldUuid String
}
```

I also want to point out that I'm not planning on using neither of `chartAsX` or `chartAsY`. So basically, I don't think I'll ever need a Chart entity when I get a Field. However, I will definitely need the field uuids every time I get a chart, and some times having the whole field entity might be useful.

My first question is, are the two fields `chartAsX` and `chartAsY` adding extra stuff to the database? I think the answer is no, but I just wanted to make sure.

My second question is, could I just remove the two `Field` properties from the Chart model and keep the `xFieldUuid` and `yFieldUuid`? My understanding is that it should work, but PrismaClient wouldn't automatically JOIN the Fields for me, so I'd have to do that manually. Again, I want to be sure I understand it.

Thanks! :)

**Comment by austincrim** [comment, marked as answer]
Hey @dgrcode, thanks for the kind words about our docs 🙂.

> My first question is, are the two fields chartAsX and chartAsY adding extra stuff to the database? I think the answer is no, but I just wanted to make sure.

There isn't anything extra added to the database, it only changes the underlying names of the relations in the database.

> My second question is, could I just remove the two Field properties from the Chart model and keep the xFieldUuid and yFieldUuid? 

If you remove the `Field` property from the `Chart` model, you won't be able to get a `Chart` and its `Fields` in a single Prisma call unless you use [`queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw). It's up to you whether that is an acceptable constraint.

Let me know if I can clarify further!

**Reply by dgrcode** [reply]
Awesome! Thanks for the fast answer. That's exactly what I wanted to know. Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
